clang/lib/CIR/Dialect/Transforms/LoweringPrepare.cpp

```diff
@@ -160,63 +160,66 @@ cir::FuncOp LoweringPreparePass::buildCXXGlobalVarDeclInitFunc(GlobalOp op) {
                                   block.begin(), std::prev(block.end()));
 
   // Register the destructor call with __cxa_atexit
-
-  assert(op.getAst() && op.getAst()->getTLSKind() == clang::VarDecl::TLS_None &&
-         " TLS NYI");
-  // Create a variable that binds the atexit to this shared object.
-  builder.setInsertionPointToStart(&theModule.getBodyRegion().front());
-  auto Handle = buildRuntimeVariable(builder, "__dso_handle", op.getLoc(),
-                                     builder.getI8Type());
-
-  // Look for the destructor call in dtorBlock
-  auto &dtorBlock = op.getDtorRegion().front();
-  mlir::cir::CallOp dtorCall;
-  for (auto op : reverse(dtorBlock.getOps<mlir::cir::CallOp>())) {
-    dtorCall = op;
-    break;
+  auto &dtorRegion = op.getDtorRegion();
+  if (!dtorRegion.empty()) {
```

Comment:
actually, this check is the only addition in this function, despite the fact the diff looks so big.
The check is added here because the call site looks like 
```
if (!ctorRegion.empty() || !dtorRegion.empty()) {
    auto f = buildCXXGlobalVarDeclInitFunc(op);
    ...
```
and we fail once `dtor` is empty